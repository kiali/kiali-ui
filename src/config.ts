import deepFreeze from 'deep-freeze';

export const config = () => {
  return deepFreeze({
    version: '0.1',
    toolbar: {
      defaultDuration: 60,
      intervalDuration: {
        60: 'Last minute',
        300: 'Last 5 minutes',
        600: 'Last 10 minutes',
        1800: 'Last 30 minutes',
        3600: 'Last hour',
        10800: 'Last 3 hours',
        21600: 'Last 6 hours',
        43200: 'Last 12 hours',
        86400: 'Last day',
        604800: 'Last 7 days',
        2592000: 'Last 30 days'
      },
      defaultPollInterval: 15000,
      pollInterval: {
        0: 'Pause',
        5000: '5 seconds',
        10000: '10 seconds',
        15000: '15 seconds',
        30000: '30 seconds',
        60000: '1 minute',
        300000: '5 minutes'
      },
      graphLayouts: {
        breadthfirst: 'Breadthfirst',
        cola: 'Cola',
        cose: 'Cose',
        dagre: 'Dagre',
        klay: 'Klay'
      }
    }
  });
};

export const KEY_CODES = { TAB_KEY: 9, ENTER_KEY: 13, ESCAPE_KEY: 27 };

export const socialLinks = [
  {
    url: 'https://github.com/kiali',
    icon: { type: 'fa', name: 'github' },
    label: 'Github'
  },
  {
    url: 'https://www.youtube.com/channel/UCcm2NzDN_UCZKk2yYmOpc5w',
    icon: { type: 'fa', name: 'youtube' },
    label: 'Youtube'
  },
  {
    url: 'https://twitter.com/KialiProject',
    icon: { type: 'fa', name: 'twitter' },
    label: 'Twitter'
  }
];
