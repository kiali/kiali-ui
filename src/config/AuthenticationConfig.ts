import { AuthConfig, AuthStrategy } from '../types/Auth';

const authenticationConfig: AuthConfig = {
  strategy: AuthStrategy.login,
  secretMissing: false
};

// Returns true if authentication strategy is either 'openshift' or 'openid'
export const isAuthStrategyOAuth = () =>
  authenticationConfig.strategy === AuthStrategy.openshift || authenticationConfig.strategy === AuthStrategy.openid;

export default authenticationConfig;
