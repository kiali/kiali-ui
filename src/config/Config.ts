import deepFreeze from 'deep-freeze';
import { UNIT_TIME, MILLISECONDS } from '../types/Common';

const conf = {
  version: '0.1',
  /** Configuration related with session */
  session: {
    /** TimeOut Session remain for warning user default 1 minute */
    timeOutforWarningUser: 1 * UNIT_TIME.MINUTE * MILLISECONDS
  },
  /** Toolbar Configuration */
  toolbar: {
    /** Duration default in 1 minute */
    defaultDuration: 1 * UNIT_TIME.MINUTE,
    /** By default refresh is 15 seconds */
    defaultPollInterval: 15 * MILLISECONDS,
    /** Options in refresh */
    pollInterval: {
      0: 'Pause',
      10000: 'Every 10s',
      15000: 'Every 15s',
      30000: 'Every 30s',
      60000: 'Every 1m',
      300000: 'Every 5m',
      900000: 'Every 15m'
    },
    /** Graphs layouts types */
    graphLayouts: {
      cola: 'Cola',
      'cose-bilkent': 'Cose',
      dagre: 'Dagre'
    }
  },
  /** About dialog configuration */
  about: {
    project: {
      url: 'https://github.com/kiali',
      iconName: 'github',
      iconType: 'fa',
      linkText: 'Find us on GitHub'
    },
    website: {
      url: 'http://kiali.io',
      iconName: 'home',
      iconType: 'fa',
      linkText: 'Visit our web page'
    }
  },
  /**  Login configuration */
  login: {
    headers: {
      'X-Auth-Type-Kiali-UI': '1'
    }
  },
  /** API configuration */
  api: {
    urls: {
      authenticate: 'api/authenticate',
      authInfo: 'api/auth/info',
      apps: (namespace: string) => `api/namespaces/${namespace}/apps`,
      app: (namespace: string, app: string) => `api/namespaces/${namespace}/apps/${app}`,
      appGraphElements: (namespace: string, app: string, version?: string) => {
        const baseUrl = `api/namespaces/${namespace}/applications/${app}`;
        const hasVersion = version && version !== 'unknown';
        const versionSuffixed = hasVersion ? `${baseUrl}/versions/${version}` : baseUrl;
        return `${versionSuffixed}/graph`;
      },
      appHealth: (namespace: string, app: string) => `api/namespaces/${namespace}/apps/${app}/health`,
      appMetrics: (namespace: string, app: string) => `api/namespaces/${namespace}/apps/${app}/metrics`,
      appDashboard: (namespace: string, app: string) => `api/namespaces/${namespace}/apps/${app}/dashboard`,
      customDashboard: (namespace: string, app: string, template: string) =>
        `api/namespaces/${namespace}/apps/${app}/customdashboard/${template}`,
      grafana: 'api/grafana',
      istioConfig: (namespace: string) => `api/namespaces/${namespace}/istio`,
      istioConfigCreate: (namespace: string, objectType: string) => `api/namespaces/${namespace}/istio/${objectType}`,
      istioConfigCreateSubtype: (namespace: string, objectType: string, objectSubtype: string) =>
        `api/namespaces/${namespace}/istio/${objectType}/${objectSubtype}`,
      istioConfigDetail: (namespace: string, objectType: string, object: string) =>
        `api/namespaces/${namespace}/istio/${objectType}/${object}`,
      istioConfigDetailSubtype: (namespace: string, objectType: string, objectSubtype: string, object: string) =>
        `api/namespaces/${namespace}/istio/${objectType}/${objectSubtype}/${object}`,
      jaeger: 'api/jaeger',
      logout: 'api/logout',
      namespaces: 'api/namespaces',
      namespacesGraphElements: `api/namespaces/graph`,
      namespaceHealth: (namespace: string) => `api/namespaces/${namespace}/health`,
      namespaceMetrics: (namespace: string) => `api/namespaces/${namespace}/metrics`,
      namespaceTls: (namespace: string) => `api/namespaces/${namespace}/tls`,
      pod: (namespace: string, pod: string) => `api/namespaces/${namespace}/pods/${pod}`,
      podLogs: (namespace: string, pod: string) => `api/namespaces/${namespace}/pods/${pod}/logs`,
      serverConfig: `api/config`,
      services: (namespace: string) => `api/namespaces/${namespace}/services`,
      service: (namespace: string, service: string) => `api/namespaces/${namespace}/services/${service}`,
      serviceGraphElements: (namespace: string, service: string) =>
        `api/namespaces/${namespace}/services/${service}/graph`,
      serviceHealth: (namespace: string, service: string) => `api/namespaces/${namespace}/services/${service}/health`,
      serviceMetrics: (namespace: string, service: string) => `api/namespaces/${namespace}/services/${service}/metrics`,
      serviceDashboard: (namespace: string, service: string) =>
        `api/namespaces/${namespace}/services/${service}/dashboard`,
      status: 'api/status',
      workloads: (namespace: string) => `api/namespaces/${namespace}/workloads`,
      workload: (namespace: string, workload: string) => `api/namespaces/${namespace}/workloads/${workload}`,
      workloadGraphElements: (namespace: string, workload: string) =>
        `api/namespaces/${namespace}/workloads/${workload}/graph`,
      workloadHealth: (namespace: string, workload: string) =>
        `api/namespaces/${namespace}/workloads/${workload}/health`,
      workloadMetrics: (namespace: string, workload: string) =>
        `api/namespaces/${namespace}/workloads/${workload}/metrics`,
      workloadDashboard: (namespace: string, workload: string) =>
        `api/namespaces/${namespace}/workloads/${workload}/dashboard`
    }
  }
};

export const config = deepFreeze(conf) as typeof conf;
