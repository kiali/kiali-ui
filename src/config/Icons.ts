import deepFreeze from 'deep-freeze';

import solidPinIcon from '../assets/img/solid-pin.png';
import hollowPinIcon from '../assets/img/hollow-pin.png';

export { solidPinIcon, hollowPinIcon };

const mutIcons = {
  istio: {
    circuitBreaker: { className: 'fa fa-bolt', type: 'fa', name: 'bolt', ascii: '\uf0e7 ' },
    missingSidecar: { className: 'fa pficon-blueprint', type: 'pf', name: 'blueprint', ascii: '\ue915 ', color: 'red' },
    mtls: { className: 'fa pf-locked', type: 'pf', name: 'locked', ascii: '\ue923 ' },
    disabledMtls: { className: 'fa fa-unlock', type: 'fa', name: 'unlock', ascii: '\uf09c ' },
    virtualService: { className: 'fa fa-code-fork', type: 'fa', name: 'code-fork', ascii: '\uf126 ' }
  },
  menu: {
    overview: 'fa fa-tachometer',
    graph: 'fa pficon-topology',
    applications: 'fa pficon-applications',
    workloads: 'fa pficon-bundle',
    services: 'fa pficon-service',
    istioConfig: 'fa pficon-template',
    distributedTracing: 'fa fa-paw'
  }
};

export const icons = deepFreeze(mutIcons) as typeof mutIcons;
