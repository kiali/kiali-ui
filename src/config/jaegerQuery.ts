import deepFreeze from 'deep-freeze';

const jaegerQueryConfig = {
  path: '/search',
  embed: {
    uiEmbed: 'uiEmbed',
    version: 'v0'
  }
};

export const jaegerQuery = () => {
  return deepFreeze(jaegerQueryConfig) as typeof jaegerQueryConfig;
};
