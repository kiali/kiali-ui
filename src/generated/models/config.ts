/* tslint:disable */
/* eslint-disable */
/**
 * Kiali
 * Kiali project, observability for the Istio service mesh
 *
 * The version of the OpenAPI document: _
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { Elements } from './elements';

/**
 *
 * @export
 * @interface Config
 */
export interface Config {
  /**
   *
   * @type {number}
   * @memberof Config
   */
  duration?: number;
  /**
   *
   * @type {Elements}
   * @memberof Config
   */
  elements?: Elements;
  /**
   *
   * @type {string}
   * @memberof Config
   */
  graphType?: string;
  /**
   *
   * @type {number}
   * @memberof Config
   */
  timestamp?: number;
}
