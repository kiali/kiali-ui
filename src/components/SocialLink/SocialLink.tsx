import * as React from 'react';
import { Icon } from 'patternfly-react';
import { socialLinks } from '../../config';

class SocialLink extends React.Component {
  render() {
    return (
      <ul className={'login-pf-page-footer-links list-unstyled'}>
        {socialLinks.map(socialMedia => (
          <li key={socialMedia.icon.name}>
            <a className={'login-pf-page-footer-link'} href={socialMedia.url} target="_blank">
              <Icon type={socialMedia.icon.type} name={socialMedia.icon.name} /> {socialMedia.label}
            </a>
          </li>
        ))}
      </ul>
    );
  }
}

export default SocialLink;
