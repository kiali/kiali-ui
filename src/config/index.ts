// Configuration

import { config } from './config';

// Icons
import { icons } from './icons';

// Logos
import { kialiLogo } from './logos';

// Paths
import { Paths } from './Paths';

// Jaeger Query
import { JAEGER_QUERY } from './jaegerQuery';

export { config, Paths, icons, kialiLogo, JAEGER_QUERY };
