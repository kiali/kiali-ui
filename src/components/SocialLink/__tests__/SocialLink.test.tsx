import * as React from 'react';
import { shallow } from 'enzyme';
import SocialLink from '../SocialLink';
import { socialLinks } from '../../../config';

const socialLinksLength = socialLinks.length;
const wrapper = shallow(<SocialLink />);

describe('#Badge render correctly with data', () => {
  it('should render badge', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain exactly the same number of links provided in the configuration', () => {
    expect(wrapper.find('li').length).toEqual(socialLinksLength);
  });

  it('should contain an Icon with link in each socialLink', () => {
    wrapper.find('li').forEach(link => {
      expect(link.find('Icon').length === 1).toBeTruthy();
      expect(link.find('a').length === 1).toBeTruthy();
    });
  });
});
