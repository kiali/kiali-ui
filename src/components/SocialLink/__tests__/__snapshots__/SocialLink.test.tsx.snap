// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Badge render correctly with data should render badge 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialLink />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          <a
            className="login-pf-page-footer-link"
            href="https://github.com/kiali"
            target="_blank"
          >
            <Icon
              name="github"
              type="fa"
            />
             
            Github
          </a>
        </li>,
        <li>
          <a
            className="login-pf-page-footer-link"
            href="https://www.youtube.com/channel/UCcm2NzDN_UCZKk2yYmOpc5w"
            target="_blank"
          >
            <Icon
              name="youtube"
              type="fa"
            />
             
            Youtube
          </a>
        </li>,
        <li>
          <a
            className="login-pf-page-footer-link"
            href="https://twitter.com/KialiProject"
            target="_blank"
          >
            <Icon
              name="twitter"
              type="fa"
            />
             
            Twitter
          </a>
        </li>,
      ],
      "className": "login-pf-page-footer-links list-unstyled",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "github",
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="login-pf-page-footer-link"
            href="https://github.com/kiali"
            target="_blank"
          >
            <Icon
              name="github"
              type="fa"
            />
             
            Github
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Icon
                name="github"
                type="fa"
              />,
              " ",
              "Github",
            ],
            "className": "login-pf-page-footer-link",
            "href": "https://github.com/kiali",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "name": "github",
                "type": "fa",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " ",
            "Github",
          ],
          "type": "a",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "youtube",
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="login-pf-page-footer-link"
            href="https://www.youtube.com/channel/UCcm2NzDN_UCZKk2yYmOpc5w"
            target="_blank"
          >
            <Icon
              name="youtube"
              type="fa"
            />
             
            Youtube
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Icon
                name="youtube"
                type="fa"
              />,
              " ",
              "Youtube",
            ],
            "className": "login-pf-page-footer-link",
            "href": "https://www.youtube.com/channel/UCcm2NzDN_UCZKk2yYmOpc5w",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "name": "youtube",
                "type": "fa",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " ",
            "Youtube",
          ],
          "type": "a",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "twitter",
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="login-pf-page-footer-link"
            href="https://twitter.com/KialiProject"
            target="_blank"
          >
            <Icon
              name="twitter"
              type="fa"
            />
             
            Twitter
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Icon
                name="twitter"
                type="fa"
              />,
              " ",
              "Twitter",
            ],
            "className": "login-pf-page-footer-link",
            "href": "https://twitter.com/KialiProject",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "name": "twitter",
                "type": "fa",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " ",
            "Twitter",
          ],
          "type": "a",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <a
              className="login-pf-page-footer-link"
              href="https://github.com/kiali"
              target="_blank"
            >
              <Icon
                name="github"
                type="fa"
              />
               
              Github
            </a>
          </li>,
          <li>
            <a
              className="login-pf-page-footer-link"
              href="https://www.youtube.com/channel/UCcm2NzDN_UCZKk2yYmOpc5w"
              target="_blank"
            >
              <Icon
                name="youtube"
                type="fa"
              />
               
              Youtube
            </a>
          </li>,
          <li>
            <a
              className="login-pf-page-footer-link"
              href="https://twitter.com/KialiProject"
              target="_blank"
            >
              <Icon
                name="twitter"
                type="fa"
              />
               
              Twitter
            </a>
          </li>,
        ],
        "className": "login-pf-page-footer-links list-unstyled",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "github",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="login-pf-page-footer-link"
              href="https://github.com/kiali"
              target="_blank"
            >
              <Icon
                name="github"
                type="fa"
              />
               
              Github
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Icon
                  name="github"
                  type="fa"
                />,
                " ",
                "Github",
              ],
              "className": "login-pf-page-footer-link",
              "href": "https://github.com/kiali",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "name": "github",
                  "type": "fa",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              " ",
              "Github",
            ],
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "youtube",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="login-pf-page-footer-link"
              href="https://www.youtube.com/channel/UCcm2NzDN_UCZKk2yYmOpc5w"
              target="_blank"
            >
              <Icon
                name="youtube"
                type="fa"
              />
               
              Youtube
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Icon
                  name="youtube"
                  type="fa"
                />,
                " ",
                "Youtube",
              ],
              "className": "login-pf-page-footer-link",
              "href": "https://www.youtube.com/channel/UCcm2NzDN_UCZKk2yYmOpc5w",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "name": "youtube",
                  "type": "fa",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              " ",
              "Youtube",
            ],
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "twitter",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="login-pf-page-footer-link"
              href="https://twitter.com/KialiProject"
              target="_blank"
            >
              <Icon
                name="twitter"
                type="fa"
              />
               
              Twitter
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Icon
                  name="twitter"
                  type="fa"
                />,
                " ",
                "Twitter",
              ],
              "className": "login-pf-page-footer-link",
              "href": "https://twitter.com/KialiProject",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "name": "twitter",
                  "type": "fa",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              " ",
              "Twitter",
            ],
            "type": "a",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
